### PR TITLE
2442 change travis to deploy with user-dashboard using git tag

### PR DESCRIPTION
**Pull request by vietthang207**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,4 +50,16 @@ This repository is using `Reactjs` and `Redux` and following their naming conven
...
 ```
 bundle exec rake master_data:media_post
-```
\ No newline at end of file
+```
+
+## Deploy to develop server
+```
+git tag -f deploy_server_develop_env_<insert env here>
+git push -f <origin/upstream> deploy_server_develop_env_<insert env here>
```
Comment (favadi) [inline README.md]: I think it is safe to just deploy all available environments of develop on new commit?
@vietthang207 please consult @andrewsource.

File: .travis/parse_tag.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+if [[ -z "$TRAVIS_TAG" ]]; then
+    echo "TRAVIS_TAG is not defined, abort."
+else
+    regex="^deploy_server_(.*)_env_(.*)$"
+    
+    if [[ $TRAVIS_TAG =~ $regex ]]; then
+        export SWAP_SERVER="${BASH_REMATCH[1]}"
+        export SWAP_ENV="${BASH_REMATCH[2]}"
+        echo server: $SWAP_SERVER env: $SWAP_ENV
```
Comment (favadi) [inline .travis/parse_tag.sh]: when there is formatting directive, using printf is always better.

Example: `printf 'server: %s env: %s' "$SWAP_SERVER" "$SWAP_ENV"`.

File: .travis/awx_job_launch.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (favadi) [inline .travis/awx_job_launch.sh]: I don't see anything trigger this script. Could you explain how does it work?

Comment (vietthang207) [reply .travis/awx_job_launch.sh]: I'll remove this temporarily until we are ready to config auto launch.

File: .travis/parse_tag.sh (hunk 2 above)
Comment (vietthang207) [reply .travis/parse_tag.sh]: I've modified  this printing.

File: README.md (hunk 1 above)
Comment (vietthang207) [reply README.md]: It is safe, it just take 1-2 mins longer to build.

Comment (favadi) [reply README.md]: Then please change the PR to only read tag and deploy for staging.

Comment (vietthang207) [reply README.md]: Sometimes we need to use a different build env in dev (i.e. mainnet/ropsten)

File: .travis/awx_job_launch.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    exit 0
+fi
+
+# VERY IMPORTANT: "prompt on launch" need to be set in the job template config in order for this API call to work.
+curl -X POST $awx_job_launch_uri \
+    --output /dev/null \
+    --write-out "HTTP response code: %{http_code}\n" \
+    --user "$awx_username:$awx_password" \
+    --header "Content-Type: application/json" \
+    --data '{ "extra_vars": {"kyber_swap_revision": "'$TRAVIS_BRANCH'"}}'
```
Comment (favadi) [inline .travis/awx_job_launch.sh]: same comment about removing extra_vars.

File: .travis.yml [hunk 5]
```diff
@@ -86,8 +77,24 @@ deploy:
...
+    local_dir: dist
+    upload_dir: $TRAVIS_COMMIT
+    skip_cleanup: true
+    on:
+      all_branches: true
+
+  - provider: script
+    script: bash .travis/awx_job_launch.sh $TRAVIS_BRANCH
+    on:
+      all_branches: true
```
Comment (favadi) [inline .travis.yml]: should be develop/staging/master only, right?

**Pull request merged**